addons/base_import/models/base_import.py

```diff
@@ -205,7 +203,8 @@ def _match_headers(self, rows, fields, options):
             for index, header in enumerate(headers)
         )
 
-    def parse_preview(self, cr, uid, id, options, count=10, context=None):
+    @api.model
+    def parse_preview(self, id, options, count=10):
```

Comment:
Do not pass id, decorate this method with @api.one, call this method with set recordset